services/terrain/src/terrain/services/permanent_id_requests.clj

```diff
@@ -38,11 +44,21 @@
              :path staging-dest})))
   data-item)
 
+(defn- validate-publish-dest
+  [{:keys [path] :as data-item}]
+  (let [publish-dest (ft/path-join (config/permanent-id-publish-dir) (ft/basename path))]
+    (when (data-info/path-exists? (config/irods-user) publish-dest)
+      (throw+ {:type :clojure-commons.exception/exists
+               :error "A folder with this name has already been published."
+               :path publish-dest})))
```

Comment:
This block might be another use case for a validator endpoint in data-info, to push the particular logic of which error codes to throw for files which already exist into it. I suppose this error message would be harder to preserve that way, though.